dvc/state.py

```diff
@@ -89,10 +88,11 @@ class State:  # pylint: disable=too-many-instance-attributes
     MAX_INT = 2 ** 63 - 1
     MAX_UINT = 2 ** 64 - 2
 
-    def __init__(self, repo):
+    def __init__(self, local_cache):
+        repo = local_cache.repo
         self.repo = repo
         self.root_dir = repo.root_dir
-        self.tree = WorkingTree(self.root_dir)
+        self.tree = local_cache.tree.work_tree
```

Comment:
@pmrowla State itself has nothing to do with the cache, but rather with the repos working tree, right? So maybe we should have `work_tree`(or wtree) in repo itself and make local cache and state use it? Just not quite obvious why state depends on local cache here.

Reply:
It seemed to me that since state is only applicable whenever there is a local cache, and used as a cache for calculating md5s which will go into the local cache, it would make more sense to have state associated with the local cache's working tree rather than with a repo tree instance. The repo doesn't actually need a working tree to function (since we can use git trees), but local cache and state only use a working tree.